dotCMS/src/main/java/com/dotcms/workflow/helper/WorkflowHelper.java

```diff
@@ -0,0 +1,519 @@
+
+import static com.dotmarketing.db.HibernateUtil.addSyncCommitListener;
+
+/**
+ * Helper for Workflow Actions
+ * @author jsanca
+ */
+public class WorkflowHelper {
+
+    private final WorkflowAPI workflowAPI;
+    private final RoleAPI     roleAPI;
+
+    private static class SingletonHolder {
+        private static final WorkflowHelper INSTANCE = new WorkflowHelper();
+    }
+
+    public static WorkflowHelper getInstance() {
+        return WorkflowHelper.SingletonHolder.INSTANCE;
+    }
+
+    private WorkflowHelper() {
+        this( APILocator.getWorkflowAPI(), APILocator.getRoleAPI());
+    }
+
+    @VisibleForTesting
+    protected WorkflowHelper(final WorkflowAPI workflowAPI,
+                             final RoleAPI     roleAPI) {
+
+        this.workflowAPI = workflowAPI;
+        this.roleAPI     = roleAPI;
+    }
+
+
+    /**
+     * Reorder the action associated to the scheme.
+     * @param workflowReorderActionStepForm WorkflowReorderBean
+     */
+    @WrapInTransaction
+    public void reorderAction(final WorkflowReorderBean workflowReorderActionStepForm,
+                              final User user)  {
+
+        WorkflowAction action = null;
+        WorkflowStep step     = null;
+
+        try {
+
+            Logger.debug(this, "Looking for the actionId: "
+                    + workflowReorderActionStepForm.getActionId());
+            action =
+                    this.workflowAPI.findAction(workflowReorderActionStepForm.getActionId(), user);
+
+            Logger.debug(this, "Looking for the stepId: "
+                    + workflowReorderActionStepForm.getStepId());
+            step =
+                    this.workflowAPI.findStep(workflowReorderActionStepForm.getStepId());
+
+            if (null == action) {
+                throw new DoesNotExistException("Workflow-does-not-exists-action");
+            }
+
+            if (null == step) {
+                throw new DoesNotExistException("Workflow-does-not-exists-step");
+            }
+
+            Logger.debug(this, "Reordering the action: " + action.getId()
+                    + " for the stepId: " + step.getId() + ", order: " +
+                    workflowReorderActionStepForm.getOrder()
+            );
+
+            this.workflowAPI.reorderAction(action, step, user,
+                    workflowReorderActionStepForm.getOrder());
+        } catch (DotDataException | DotSecurityException | AlreadyExistException e) {
+
+            Logger.error(this, e.getMessage(), e);
+            throw new DotWorkflowException(e.getMessage(), e);
+        }
+    }  // reorderAction.
+
+    /**
+     * Deletes the step
+     * @param stepId String
+     */
+    @WrapInTransaction
+    public void deleteStep(final String stepId) {
+
+        WorkflowStep workflowStep = null;
+
+        try {
+
+            Logger.debug(this, "Looking for the stepId: " + stepId);
+            workflowStep = this.workflowAPI.findStep(stepId);
+        } catch (Exception e) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)